### PR TITLE
#88 improve evaluation period behaviour

### LINKED ISSUE #88: Change the info shown when your course has not been yet evaluated on SignETS
**Is your feature request related to a problem? Please describe.**
When I see my courses under the student tab during the period of professor reviewing, I see all the course that I didn't yet review as 0%.

image

**Describe the solution you'd like**
I would like to see a warning or info icon instead. 

**Describe alternatives you've considered**
A popup when It's clicked on it that tell you to go review your professor on SignETS would also be nice. Can be combined with the other solution.

**Additional Information**
To access if a course is pending an evaluation, we need to call lireEvaluationCours. The response from the API is like this:
```xml
<soap:Body>
      <lireEvaluationCoursResponse xmlns="http://etsmtl.ca/">
          <lireEvaluationCoursResult>
              <erreur />
              <listeEvaluations>
                  <EvaluationCours>
                      <Sigle>MAT265</Sigle>
                      <Groupe>09</Groupe>
                      <Enseignant>Hilal, Saïd</Enseignant>
                      <DateDebutEvaluation>2021-03-19T00:00:00</DateDebutEvaluation>
                      <DateFinEvaluation>2021-03-28T23:59:00</DateFinEvaluation>
                      <TypeEvaluation>Cours</TypeEvaluation>
                      <EstComplete>true</EstComplete>
                  </EvaluationCours>
                  <EvaluationCours>
                      <Sigle>MAT265</Sigle>
                      <Groupe>09</Groupe>
                      <Enseignant>Hilal, Saïd</Enseignant>
                      <DateDebutEvaluation>2021-03-19T00:00:00</DateDebutEvaluation>
                      <DateFinEvaluation>2021-04-01T23:59:00</DateFinEvaluation>
                      <TypeEvaluation>Cours</TypeEvaluation>
                      <EstComplete>false</EstComplete>
                  </EvaluationCours>
                  <EvaluationCours>
                      <Sigle>ELE105</Sigle>
                      <Groupe>01</Groupe>
                      <Enseignant>Constantin, Nicolas</Enseignant>
                      <DateDebutEvaluation>2021-03-19T00:00:00</DateDebutEvaluation>
                      <DateFinEvaluation>2021-03-28T23:59:00</DateFinEvaluation>
                      <TypeEvaluation>Cours</TypeEvaluation>
                      <EstComplete>true</EstComplete>
                  </EvaluationCours>
                  <EvaluationCours>
                      <Sigle>ELE105</Sigle>
                      <Groupe>01</Groupe>
                      <Enseignant>Constantin, Nicolas</Enseignant>
                      <DateDebutEvaluation>2021-03-19T00:00:00</DateDebutEvaluation>
                      <DateFinEvaluation>2021-04-01T23:59:00</DateFinEvaluation>
                      <TypeEvaluation>Cours</TypeEvaluation>
                      <EstComplete>false</EstComplete>
                  </EvaluationCours>
                  <EvaluationCours>
                      <Sigle>CHM131</Sigle>
                      <Groupe>11</Groupe>
                      <Enseignant>Baril, Pierre</Enseignant>
                      <DateDebutEvaluation>2021-03-19T00:00:00</DateDebutEvaluation>
                      <DateFinEvaluation>2021-03-28T23:59:00</DateFinEvaluation>
                      <TypeEvaluation>Cours</TypeEvaluation>
                      <EstComplete>true</EstComplete>
                  </EvaluationCours>
                  <EvaluationCours>
                      <Sigle>CHM131</Sigle>
                      <Groupe>11</Groupe>
                      <Enseignant>Baril, Pierre</Enseignant>
                      <DateDebutEvaluation>2021-03-19T00:00:00</DateDebutEvaluation>
                      <DateFinEvaluation>2021-04-01T23:59:00</DateFinEvaluation>
                      <TypeEvaluation>Cours</TypeEvaluation>
                      <EstComplete>false</EstComplete>
                  </EvaluationCours>
              </listeEvaluations>
          </lireEvaluationCoursResult>
      </lireEvaluationCoursResponse>
  </soap:Body>
```
There is two evaluation for the same course because of the pandemic, there is the normal evaluation and the one to evaluate the remote work. So as soon as one of the `<EvaluationCours><EstComplete>` field is false it shouldn't. On the other hand, I think currently there is a bug in the API, because in SignETS it is clearly stated that the second evaluation is optional. But as you can see from the LireElementsEvaluation, my grade is still at 0. It's not really a big problem right now I think, as in the next 2-3 semester there won't even  have a second evaluation, optional or not.

```xml
<soap:Body>
    <listeElementsEvaluationResponse xmlns="http://etsmtl.ca/">
        <listeElementsEvaluationResult>
            <erreur />
            <noteACeJour>0,0</noteACeJour>
            <scoreFinalSur100>0,0</scoreFinalSur100>
            <moyenneClasse>35,9</moyenneClasse>
            <ecartTypeClasse>9,9</ecartTypeClasse>
            <medianeClasse>38,4</medianeClasse>
            <rangCentileClasse>0,0</rangCentileClasse>
            <noteACeJourElementsIndividuels>0,0</noteACeJourElementsIndividuels>
            <noteSur100PourElementsIndividuels>0,0</noteSur100PourElementsIndividuels>
            <tauxPublication>50,0</tauxPublication>
            <liste>
                <ElementEvaluation>...</ElementEvaluation>
                <ElementEvaluation>...</ElementEvaluation>
                <ElementEvaluation>...</ElementEvaluation>
            </liste>
        </listeElementsEvaluationResult>
    </listeElementsEvaluationResponse>
</soap:Body>
```

### PR DESCRIPTION
**Pull request by apomalyn**
Will closes #88 

This PR improves the application behaviour when we are in the course evaluation. (Screenshots will come shortly :smile:)
When a course evaluation isn't completed and we are in the evaluation period the grade button displays 'N/A'.

I also added the name of the teacher on the AppBar.

<details>

</details>

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ description: The 4th generation of ÉTSMobile, the main gateway between the Éco
 # pub.dev using `pub publish`. This is preferred for private packages.
 publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 
-version: 4.5.3+1
+version: 4.6.3+1
```
Comment (MysticFragilist) [inline pubspec.yaml:8]: this should be 4.6.0 😄

File: test/managers/course_repository_test.dart [hunk 2]
```diff
@@ -4,6 +4,7 @@ import 'dart:convert';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:intl/intl.dart';
 import 'package:mockito/mockito.dart';
+import 'package:notredame/core/models/course_evaluation.dart';
```
Comment (MysticFragilist) [inline test/managers/course_repository_test.dart:7]: should go in the models import

**Review by MysticFragilist — CHANGES_REQUESTED** [review]

File: lib/l10n/intl_en.arb [hunk 3]
```diff
@@ -69,7 +69,7 @@
   "schedule_error_no_event_found": "No event found",
 
   "grades_title": "Grades",
-  "grades_msg_no_grades": "No grade available\nPull to refresh.",
+  "grades_msg_no_grades": "No grade available",
```
[the remark this answers is not in the document]
Comment (MysticFragilist) [reply lib/l10n/intl_en.arb]: oh nevermind you changed it my bad ^^

File: pubspec.yaml (hunk 1 above)
Comment (apomalyn) [reply pubspec.yaml:8]: My bad... 😅

**Comment by MysticFragilist** [comment]
I'll test it right now and if it's all good, I'll LGTM right away!

**Comment by camillebrulotte** [comment]
Good idea for the teachers name !

**Review by MysticFragilist — APPROVED** [review]
LGTM 🚀

**Review by camillebrulotte — APPROVED** [review]
LGTM :D

**Pull request merged**